lib/spack/spack/compiler.py

```diff
@@ -707,6 +778,28 @@ def compiler_environment(self):
             os.environ.update(backup_env)
 
 
+class CompilerLang:
+    CC = "cc"
+    CXX = "cxx"
+    FC = "fc"
+    F77 = "f77"
+    ext_for_lang = {CC: "c", CXX: "cxx", FC: "f", F77: "f77"}
```

Comment:
?

Reply:
Cruft from an earlier pass. This PR wasn't actually ready to go in when you halted progress on it, and I don't like to remove code until I'm 100% sure I'm not going to need it again.

Reply:
@johnwparent Didn't we agree yesterday that I would review this PR asap, as soon as you ping me in here, when you thought that was ready for review? I saw the ping, and I reviewed the PR :thinking: Now you say it's not ready... fine. Ping me when it's ready and I'll re-review.

Reply:
> Apologies if that wasn't what you had intended, we should have clarified what this review would be for, I did not mean to mislead you as to the state of this PR, again, my apologies.

No worries, misunderstanding happens (often, apparently, in this PR :grimacing: ). I'm happy to re-review when ready.

Reply:
> misunderstanding happens (often, apparently, in this PR 😬 )

I think we're on the same page now!